### DISCUSSION TITLE
Wheel slip on HSRB's base (hardware problem)

### DISCUSSION THREAD

**Discussion by arata** [HumanSupportRobot/QandA#29] [Show and tell]
This is not a question, but I will share how we fixed it.
First, We noticed a strange behavior on the robot's navigation; finally, it turned out to be a hardware problem, as shown in the picture.

How to fix it.
1. Remove the side cover of the base.
2. Remove the battery from HSR.
3. Lay the HSR down and remove the bottom base cover.
4. Remove the motor mounts and timing belt.
5. Then you will see that the set screw on the side of the pulley is loose.
6. Insert the pulley in the correct position on the motor shaft and retighten.
7. Put the timing belt back on and reattach the motor.
8. Then, reattached the base and side covers to the HSR.

I hope this is helpful to anyone else experiencing the same problem.

![IMG_20221121_200845](https://user-images.githubusercontent.com/28821947/203700411-84072772-22d8-4dd0-b7ee-87c14a30bb52.jpg)

**Comment by yani-rl-ai** [comment]
@arata thank you for your information, can you describe what kind of strange behavior from your navigation ? because we still experiencing the strange behavior during mapping, the robot always shaking in rviz during mapping.

**Comment by arata** [comment]
@ROSNovice  
It was a long time ago, so I don't remember much, but it was something like this.
For example, I simply give the command 2 meters forward "omni_base.go_rel(2, 0, 0)" and HSR moves diagonally forward while heading for the goal position, even though there is no obstacle in the area.
After that, I thought it was strange, so I publish a speed command to the topic "/hsrb/command_velocity" x directionally in the forward direction, but it moved diagonally forward and also same with joystick operation. so I was convinced that it was strange.

I hope this will be helpful.
